### PR TITLE
[PBI-1976] Add Conviva error reporting

### PR DESCRIPTION
**Pull request by dchauOoyala**

### CODE REVIEW

File: test/unit-tests/testConviva.js [hunk 1]
```diff
@@ -864,4 +864,118 @@ describe('Analytics Framework Conviva Plugin Unit Tests', function() {
...
+    });
+    var simulator = Utils.createPlaybackSimulator(plugin);
+    var embedCode = "testEmbedCode";
+    var title = "testTitle";
+    simulator.simulatePlayerLoad({
+      embedCode: embedCode,
+      title: title,
+      duration: 60000
+    });
+    simulator.simulateVideoElementCreated("http://testStreamUrl");
```
Comment (aeng7) [inline test/unit-tests/testConviva.js]: Can you pull and merge from master? There's a new function called startPlayer to simulate the video element creation and a new requirement to simulate the INITIAL_PLAYBACK_REQUESTED message.

File: test/unit-tests/testConviva.js [hunk 2]
```diff
@@ -864,4 +864,118 @@ describe('Analytics Framework Conviva Plugin Unit Tests', function() {
...
```
Comment (aeng7) [inline test/unit-tests/testConviva.js]: See above comment to use the startPlayer function.

File: test/unit-tests/testConviva.js [hunk 3]
```diff
@@ -864,4 +864,118 @@ describe('Analytics Framework Conviva Plugin Unit Tests', function() {
...
```
Comment (aeng7) [inline test/unit-tests/testConviva.js]: See above comment to use the startPlayer function.

File: test/unit-tests/testConviva.js [hunk 4]
```diff
@@ -864,4 +864,118 @@ describe('Analytics Framework Conviva Plugin Unit Tests', function() {
...
```
Comment (aeng7) [inline test/unit-tests/testConviva.js]: See above comment to use the startPlayer function.

File: test/unit-tests/testConviva.js [hunk 5]
```diff
@@ -864,4 +864,118 @@ describe('Analytics Framework Conviva Plugin Unit Tests', function() {
...
```
Comment (aeng7) [inline test/unit-tests/testConviva.js]: See above comment to use the startPlayer function.

File: test/unit-tests/testConviva.js [hunk 6]
```diff
@@ -997,11 +1010,12 @@ describe('Analytics Framework Conviva Plugin Unit Tests', function() {
...
-    simulator.simulateVideoElementCreated("http://testStreamUrl");
+    startPlayer(simulator, "http://testStreamUrl");
 
     var errorCode = "authorizationErrorCode";
     var errorMessage = "authorizationErrorMessage";
+    var errorString = createErrorString(errorCode, errorMessage);
     simulator.simulateAuthorizationError(errorCode, errorMessage);
-    expect(Conviva.currentPlayerStateManager.errorSent).toBe(errorMessage);
+    expect(Conviva.currentPlayerStateManager.errorSent).toBe(errorString);
   });
```
Comment (aeng7) [inline test/unit-tests/testConviva.js]: Can we add a unit test for an empty error message?

**Comment by aeng7** [comment]
LGTM
